mediaflow_proxy/extractors/vavoo.py

```
@@ -0,0 +1,147 @@
+            async with httpx.AsyncClient(timeout=20) as client:
+                resp = await client.post("https://vavoo.to/mediahubmx-resolve.json", json=data, headers=headers)
+                resp.raise_for_status()
+                result = resp.json()
+                if isinstance(result, list) and result and result[0].get("url"):
+                    return result[0]["url"]
+                elif isinstance(result, dict) and result.get("url"):
+                    return result["url"]
+                else:
+                    return None
+        except Exception as e:
+            raise ExtractorError(f"Vavoo resolution failed: {str(e)}") 
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)